mne/viz/utils.py

```diff
@@ -3188,18 +3191,17 @@ def _plot_psd(inst, fig, freqs, psd_list, picks_list, titles_list,
                     ch_types_used=ch_types_used, selectable=True, psd=True,
                     line_alpha=line_alpha, nave=None, time_unit='ms',
                     sphere=sphere)
-    for ii, ax in enumerate(ax_list):
+    for ii, (ax, xlabel) in enumerate(zip(ax_list, xlabels_list)):
         ax.grid(True, linestyle=':')
         if xscale == 'log':
             ax.set(xscale='log')
             ax.set(xlim=[freqs[1] if freqs[0] == 0 else freqs[0], freqs[-1]])
             ax.get_xaxis().set_major_formatter(ScalarFormatter())
         else:
             ax.set(xlim=(freqs[0], freqs[-1]))
-        if make_label:
-            if ii == len(picks_list) - 1:
-                ax.set_xlabel('Frequency (Hz)')
-            ax.set(ylabel=ylabels[ii], title=titles_list[ii])
+        if xlabel:
+            ax.set_xlabel('Frequency (Hz)')
```

Comment:
You errantly killed the `ylabels` setting

Reply:
fixed

Reply:
I think this still needs to be nested in the `if make_label` code

IIRC I had it this way in the commit I pushed to your branch. You also removed a `_check_option('xscale', xscale, ('log', 'linear'))` that I had put in, was that intentional? If not, please restore it as it takes care of the problem where `xscale` could be anything.